core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -797,17 +797,11 @@ class WorkspaceService(protected val userInfo: UserInfo,
                         val newAttrs = sourceWorkspaceContext.attributes ++ destWorkspaceRequest.attributes
                         traceDBIOWithParent("withNewWorkspaceContext (cloneWorkspace)", parentSpan) { s1 =>
                           withNewWorkspaceContext(destWorkspaceRequest.copy(authorizationDomain = Option(newAuthDomain), attributes = newAttrs), destBillingProject, sourceBucketNameOption, dataAccess, s1) { destWorkspaceContext =>
-                            dataAccess.entityQuery.copyAllEntities(sourceWorkspaceContext, destWorkspaceContext) andThen
-                              dataAccess.methodConfigurationQuery.listActive(sourceWorkspaceContext).flatMap { methodConfigShorts =>
-                                val inserts = methodConfigShorts.map { methodConfigShort =>
-                                  dataAccess.methodConfigurationQuery.get(sourceWorkspaceContext, methodConfigShort.namespace, methodConfigShort.name).flatMap { methodConfig =>
-                                    dataAccess.methodConfigurationQuery.create(destWorkspaceContext, methodConfig.get)
-                                  }
+                              dataAccess.entityQuery.cloneEntitiesToNewWorkspace(sourceWorkspaceContext.workspaceIdAsUUID, destWorkspaceContext.workspaceIdAsUUID) andThen  {
+                                dataAccess.methodConfigurationQuery.clone(sourceWorkspaceContext.workspaceIdAsUUID, destWorkspaceContext.workspaceIdAsUUID) flatMap {
+                                  _ => DBIO.successful((sourceWorkspaceContext, destWorkspaceContext))
```

Comment:
What you have here looks pretty good. Keeping this syntax, my only suggestion is to replace the last `flatMap` with another `andThen` for consistency, as below. This is probably the best approach to take, since this area of the code relies on lots of indenting.
```scala                              
 dataAccess.entityQuery.cloneEntitiesToNewWorkspace(sourceWorkspaceContext.workspaceIdAsUUID, destWorkspaceContext.workspaceIdAsUUID) andThen  {
     dataAccess.methodConfigurationQuery.clone(sourceWorkspaceContext.workspaceIdAsUUID, destWorkspaceContext.workspaceIdAsUUID) andThen {
         DBIO.successful((sourceWorkspaceContext, destWorkspaceContext))
     }
 }
```

But personally, I don't like all the indenting and I find `for … yield` easier to read. Here's what that code might look like:
```scala
 for {
     _ <- dataAccess.entityQuery.cloneEntitiesToNewWorkspace(sourceWorkspaceContext.workspaceIdAsUUID, destWorkspaceContext.workspaceIdAsUUID)
     _ <- dataAccess.methodConfigurationQuery.clone(sourceWorkspaceContext.workspaceIdAsUUID, destWorkspaceContext.workspaceIdAsUUID)
 } yield {
     DBIO.successful((sourceWorkspaceContext, destWorkspaceContext))
 }
```

a final option would be to use `DIBIO.seq`, like so:
```scala              
 val copyOperations =  DBIO.seq(dataAccess.entityQuery.cloneEntitiesToNewWorkspace(sourceWorkspaceContext.workspaceIdAsUUID, destWorkspaceContext.workspaceIdAsUUID),
     dataAccess.methodConfigurationQuery.clone(sourceWorkspaceContext.workspaceIdAsUUID, destWorkspaceContext.workspaceIdAsUUID))
                            
 copyOperations andThen
     DBIO.successful((sourceWorkspaceContext, destWorkspaceContext))
```

these are all pretty much equivalent. There's possibly some miniscule advantage to the `DBIO.seq` approach because that doesn't bother storing the result of any of its actions, whereas the `andThen`s in the first approach store a result … but the result sets are super small anyway. I'd pick whichever you think makes the code most readable.

Reply:
Cool, I appreciate these explanations!